### PR TITLE
feat: upgrade golangci-lint

### PR DESCRIPTION
**Pull request by fengshunli**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -44,7 +43,7 @@ jobs:
       - name: Lint golang code
         uses: golangci/golangci-lint-action@v3
         with:
-          version: ${{ env.GOLANGCI_VERSION }}
+          version: latest
```
Comment (rambohe-ch) [inline .github/workflows/ci.yaml:46]: why not version v1.51.1 is used here?

**Comment by rambohe-ch** [comment]
@fengshunli please fix unit tests error. you can click the link: https://github.com/openyurtio/openyurt/actions/runs/4219426618/jobs/7324769786 to check the error message.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (fengshunli) [reply .github/workflows/ci.yaml:46]: It is convenient to use latest without modifying it every time

Comment (rambohe-ch) [reply .github/workflows/ci.yaml:46]: @fengshunli In file `Makefile`, you will install golangci-lint version v1.51.1, but here latest version is used, so installed version is not used.

Comment (fengshunli) [reply .github/workflows/ci.yaml:46]: Makefile has been updated @rambohe-ch

**Pull request closed without merging**